### ISSUE TITLE
[SUGGESTION]: Atheme setting non-standard usermodes to remove flood and channel limits (+c/+f)

### ISSUE THREAD

**Issue by blitmap** [atheme/atheme#137]
Hi!

Premise: CIA and bots like it use multiple instances and IPs even to get around channel limits and sometimes flood limits.  CIA was a trusted service but still probably needed multiple instances to deal with the massive amount of input it received from the networks it lurked on.  Regardless, a bot of smaller responsibility or simply a trusted user might need to be on more than the common 60 max channels or need to send out huge amounts of text without being throttled by the IRCd (without being a limitless oper).

I thought of 2 ways to do this:

1) NickServ sets +r and the IRCd responds (if configured) by also disabling the channel and/or flood limit on the registered user (for some networks this may be the minimal amount of trust required).  This doesn't need anything from Atheme -- we're trying to do this in Insp but we're still working out a clean way of tracking unlimited users (and maybe adding an oper command for removing these limits at runtime).

2)  Atheme could set +c on the user for disabling channel limits and +f on the user for disabling flood limits.  Opers could set a property on designated user accounts for disabling channel and/or flood limits and have NickServ set +c and/or +f once the user is authed.  I think this might be a tad sexier than the first schenario.

Even if this idea sounds ridiculous to some I would love to hear feedback.  I believe SaberUK is working toward doing away with a global channel limit for some users (per configuration or a runtime oper command) but it isn't quite clean yet.  This suggestion is ahead of its need but I wanted to keep Atheme in the loop on this idea and be open to collaboration.

Hope I didn't bore or annoy with this posting, this is maybe an IRCv3 proposal?

TL;DR +c usermode for channel unlimited, +f usermode for flood unlimited -- with nickserv helping commands to set these properties on users upon auth (and then the IRCd does its own thing)
